src/BTCPayServer.Lightning.CLightning/CLightningClient.cs

```diff
@@ -281,9 +281,13 @@ private async Task<PayResponse> PayAsync(string bolt11, PayInvoiceParams payPara
                     FeeAmount = response.AmountSent - response.Amount
                 });
             }
-            catch (LightningRPCException ex) when (ex.Code == CLightningErrorCode.ROUTE_NOT_FOUND || ex.Code == CLightningErrorCode.STOPPED_RETRYING)
+            catch (LightningRPCException ex)
             {
-                return new PayResponse(PayResult.CouldNotFindRoute);
+                var result =
+                    ex.Code == CLightningErrorCode.ROUTE_NOT_FOUND || ex.Code == CLightningErrorCode.STOPPED_RETRYING
+                        ? PayResult.CouldNotFindRoute
+                        : PayResult.Error;
```

Comment:
I  don't really like it, this completely obfuscate the real problem.

It should throw an exception if it's something we never saw.

Reply:
Updated in [c5b44a9](https://github.com/btcpayserver/BTCPayServer.Lightning/pull/72/commits/c5b44a91e9b85d671f7a62a89b027d4c2a06830f). 

I came across this when working on self-payments. The error that got reported was unspecific and now we are handling it with a better return result and message as well. Also added tests for the self-payment handling in [b66acb2](https://github.com/btcpayserver/BTCPayServer.Lightning/pull/72/commits/b66acb23c5f99abfe82c27190fbead20bff05d4e).